back/src/forms/mutations/save-form.ts

```diff
@@ -1,158 +1,38 @@
 import {
-  flattenFormInput,
-  flattenTemporaryStorageDetailInput,
-  expandFormFromDb
-} from "../form-converter";
-import { getReadableId } from "../readable-id";
-import {
-  FormUpdateInput,
-  FormCreateInput,
-  Status,
-  prisma
-} from "../../generated/prisma-client";
-import { UserInputError } from "apollo-server-express";
-import { MutationSaveFormArgs, Form } from "../../generated/graphql/types";
-
-/**
- * Custom exception thrown when trying to set temporaryStorageDetail without
- * recipient.isTempStorage = true
- */
-class BadTempStorageInput extends UserInputError {
-  constructor() {
-    super(
-      "Vous ne pouvez pas préciser d'entreposage provisoire" +
-        " sans spécifier recipient.isTempStorage = true"
-    );
-  }
-}
+  MutationSaveFormArgs,
+  Form,
+  ResolversParentTypes
+} from "../../generated/graphql/types";
+import { GraphQLContext } from "../../types";
+import { updateForm } from "./updateForm";
+import { createForm } from "./createForm";
 
 export async function saveForm(
-  userId: string,
-  { formInput }: MutationSaveFormArgs
+  _: ResolversParentTypes["Mutation"],
```

Comment:
perso je trouve assez bizarre de se trimbaler un paramètre inutilisé `_`

Reply:
Oui c'est vrai, le parent sert très rarement voir jamais pour les mutations 😕 c'est un peu inévitable vu que c'est passé par `apollo-server` et je trouve ça plus facile à maintenir quand tout est au même endroit, plutôt que d'avoir à rester attentif au code qui est dans 2 fichiers différents (`resolvers.ts` et `saveForm.ts`). Ça permet aussi une certaine cohérence et diminue le boilerplate.

Mais ce n'est que mon avis et ce n'est pas forcément un point auquel je donne beaucoup d'importance. On pourrait aussi très bien dire que toutes mutations ont une signature sans le parent (qui aura été omit dans `resolvers.ts` avant d'appeller la fonction correspondante).